src/node.ts

```diff
@@ -39,39 +33,29 @@ export class BPTreeNode {
 			let leaf = size < 0;
 			let absSize = Math.abs(size);
 
-			console.log(`Size: ${size}, Leaf: ${leaf}`);
-
-			this.pointers = new Array(absSize + (leaf ? 0 : 1))
+			node.pointers = new Array(absSize + (leaf ? 0 : 1))
```

Comment:
Since we iterate over the keys down below, there's no need to preallocate the arrays here. These can be removed